### ISSUE TITLE
Deprecate :onyx/restart-pred-fn

### ISSUE THREAD

**Issue by lbradstreet** [onyx-platform/onyx#485]
TODO
* Add deprecations feature where schema checks for any deprecations via the information model, and provides a nice error. This will require `:deprecation-version`, and `:deprecation-doc` keys in the information model.
* Remove the feature

**Comment by lbradstreet** [comment]
Needs a fix for #491 first

**Comment by MichaelDrogalis** [comment]
Started. Sample exception:

```
clojure.lang.ExceptionInfo: Task :identity failed validation. Error: {:onyx/restart-pred-fn {:cause ":onyx/restart-pred-fn has been removed from Onyx. A more general and powerful feature has been added instead, named Lifecycle Exceptions. See the docs for :lifecycle/handle-exception to switch over.", :data {:onyx/restart-pred-fn :onyx.windowing.basic-windowing-crash-test/restartable?}, :deprecated-version "0.8.9"}}
```
